src/OpenTelemetry/Trace/TracerProviderSdk.cs

```diff
@@ -220,6 +221,32 @@ protected override void Dispose(bool disposing)
             base.Dispose(disposing);
         }
 
+        protected override bool OnShutdown(int timeoutMilliseconds)
+        {
+            bool? result;
+            var sw = Stopwatch.StartNew();
+            this.listener?.Dispose();
```

Comment:
@CodeBlanch I suspect it might cause the same issue as mentioned in Line216, need your input here.

Reply:
@utpilla Can we move the disposal of the listener to after the processor(s) shutdown? I'll have to test it, but I suspect @reyang  is correct.

Reply:
@CodeBlanch I have moved listener dispose to after processor dispose.

Reply:
Need to assign the listener to null after Dispose is called?

Reply:
Activity listener is currently defined as a readonly field. We would have to change that to assign it to null in the Shutdown method.

Reply:
Not sure whether should we prevent multiple calls to ActivityListener.Dispose or not. If one calls Dispose() on TracerproviderSdk multiple times, we are not doing any extra checks and are simply calling listener's dispose.  
@utpilla Since this pattern already exists before this PR, we can merge this PR, but please open a new issue to track Dispose Improvements for TracerProvider.

Reply:
I created this Issue #1505 to track it.